### DISCUSSION TITLE
Required fields in issue template don't work in organization repositories

### DISCUSSION THREAD

**Discussion by fmal** [community/community#43859] [Enterprise]
### Select Topic Area

Bug

### Body

The same issue template used in a personal repository and in an enterprise org repo behaves differently.

Fields marked as required using

```
   validations:
      required: true
```

are correctly marked and validated as required in personal repositories (example https://github.com/michalandrzejczak/issue-templates-test/issues/new/choose), but the very same template moved to enterprise organization repo results in all fields being optional (no asterisks next to field labels and no validation)

Llooks as if validations were completely ignored in enterprise org repos.

**Comment by codebydant** [comment]
Same problem in my organization

**Reply by codebydant** [reply]
Related to https://github.com/community/community/discussions/45084

**Comment by azenMatt** [comment]
Hi, this is by design and is documented here - https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms#about-yaml-syntax-for-issue-forms

Would you mind sharing what forms you were looking to create with required fields to help the team better understand the use case?

**Reply by codebydant** [reply]
@azenMatt In my case, we have a form like this:

![Screenshot from 2023-02-09 06-25-47](https://user-images.githubusercontent.com/115469901/217799969-ea679dbb-d199-4e24-b5da-a374540ec8eb.png)

The **describe bug** is optional and sometimes my team does not complete the form due to laziness

**Reply by codebydant** [reply]
We want to make sure that good practices are followed and when people aren't pushed to do it they just don't do it.

**Reply by d12frosted** [reply]
Would also want to be able to mark some of the fields as required. What makes private organisation repositories different from public repositories? We also have issue reporting flows that everyone must honour. People are lazy and skip stuff when nothing enforces them :)

**Reply by gualterandre** [reply]
We also need this. Making fields mandatory are a must have for our private forms.

**Reply by thoutbeckers** [reply]
@azenMatt can you perhaps share the underlying design decision why GitHub made this division between public and private? As in, what use case did GitHub foresee in public issues that do not exist in private ones?

Perhaps this will teach us something, e.g. maybe GitHub sees a better alternative specifically for private repos?

**Reply by datfinesoul** [reply]
@azenMatt , I agree with @thoutbeckers' comment. Understanding why GitHub thought the need for required fields only exists for public repos is not really clearly explained in the documentation.  I understand that it's a current limitation, but it does reduce the usefulness of these forms for internal workflows.  Empty form submissions are just to easily done, and ultimately just generate issue spam.

**Reply by JSMike** [reply]
@azenMatt What can be done to enable this for enterprise? I'm not sure why this wouldn't be a setting for organizations to choose, especially as organizations are paying to use the platform. This would be a very impactful feature for my org.

Has anyone come up with an action to enforce, until we can enable this for enterprise?

Our use case is internal library support.

**Reply by sky-lackner** [reply]
@azenMatt adding my request to enable this feature for private repos. This would be extremely useful to make sure devs have the necessary info to actually act on and respond to a created issue.

**Reply by Timmmy-nlb** [reply]
As I and many others have already written: we want this functionality!
https://github.com/orgs/community/discussions/51500

For the Apache Fondation it was finally also allowed!

**Reply by twisterdotcom** [reply]
@azenMatt - this is something I check consistently. I and many others have laid out reasons here: https://github.com/orgs/community/discussions/4264#discussioncomment-6071942.

I continually wonder _why_ the decision to enforce the required fields wasn't carried over, yet it's still something we can add to a form is. Is there any possibility this will be coming soon at all?

>does not complete the form due to laziness

These forms should help with exactly this —otherwise, they might as well be markdown templates. There's literally no point in them at the moment in our repos. It's just a completely wasted feature. I cannot describe how much benefit this would have to our internal support teams triaging bug reports written by first and second line.

**Reply by sgravrock** [reply]
I've been trying to work around the lack of a preview feature by testing my issue template changes in a private repo before inflicting them on real users. (Call me old fashioned but I don't think it's a good idea to put un-tested YAML changes into production.) It's really unfortunate that the exact same issue template does different things in different repos.

**Reply by dboune** [reply]
This severely degrades the utility of issue forms in an org. The ability to require a field would have significantly eased the work required to chase people around asking for more information.

While it is true that the same users creating an issue can edit the issue as markdown without the form, a carefully considered set of required fields and a restriction to prevent initial creation using plain markdown would make a daily difference.

**Reply by Timmmy-nlb** [reply]
The level of disinterest towards its (business) customers and the inaccessibility of rational arguments is quite impressive/pity.
That people are not given the freedom to decide for themselves how they act internally, just to enforce their own agenda/world view...

**Comment by mazlan** [comment]
@azenMatt - We also need required validations in our org for at least the Internal repositories. Especially as org admins, we can use GitHub issue templates to intake certain support requests and as people may miss out on filling required information this adds extra toil to reject such requests and going back to team members explaining what they missed, how to actually fill in the form etc. Not sure why this would be by design as it is already an optional attribute for public repos so why not make it the same for internal/private repos? Example of one such template:
![image](https://user-images.githubusercontent.com/1832090/228317320-e4042810-2106-46a5-a1e5-0829bb074b27.png)

**Comment by BenjaminEngeset** [comment]
Were you able to find a workaround @fmal? I have the same issue and it's also an issue with aplying automatic labels by using:
```yml
labels: ["Bug Report"]
```

This is also something that works great when it's not under an enterprise plan.

**Reply by Timonwa** [reply]
My labels work fine though, without any issues. Mores when the labels that have emoji in them.

**Reply by skylerjokiel** [reply]
I found that labels only work if the label already exists in the repo. If it doesn't exist it will not be added.

**Reply by mtfunder** [reply]
I found out today that not only does the label need to be in the repo that you are creating the issue in, it needs to be in the orgs .github repo as well... Took forever to figure that out. You also have to give anyone you want to add as a template assignee write access to the .github repo....

**Comment by jefferyjob** [comment]
1. There is no clear indication in the issue form document that private repositories cannot use form validation. I hope the document can be completed. At first, I thought it was a problem with my configuration and searched for it for a long time, 😁 
2. We hope to have a private repository that can also support issue form validation

**Comment by rantoniuk** [comment]
@jefferyjob yes, there is, see the Note: at the last section of https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms#about-yaml-syntax-for-issue-forms.
 
@azenMatt point me to one person that thinks this would by design be useful only for public organisations. I cannot imagine you need an example of use cases for this, it's like saying we need to provide usecases for existence of Github Issues 😆 unless you want us to go back to Jira... 🚶🏻‍♂️ 

Selecting whether the issue is for DEV or PROD, requiring the reporter to enter the app version string against which the bug occurred, etc. etc.

**Comment by Timmmy-nlb** [comment]
Good News: "required" is now working in Internal Repos!
https://github.com/orgs/community/discussions/148713#discussioncomment-12226996
